src/adaPets.js

```diff
@@ -9,29 +9,56 @@ const BASE_URL = 'http://localhost:3000/pets/';
 
 // Option functions.
 const listPets = () => {
-  // Fill out as part of Wave 1.
+  axios.get(BASE_URL)
+    .then((response) => {
+      setResult(response.data);
+    })
+    .catch((error) => {
+      setError(error.message);
+  });
 };
 
 const showDetails = (selectedPetId) => {
   if (!selectedPetId) {
     setError("You tried to show details for a pet without selecting it!");
     
-  }
-
-  // Fill out as part of Wave 2.
+  } else {
+    axios.get(BASE_URL + selectedPetId)
+      .then((response) => {
+        setResult(response.data);
+      })
+      .catch((error) => {
+        setError(error.message);
+    });
+  };
 };
 
 const removePet = (selectedPetId) => {
   if (!selectedPetId) {
     setError("You tried to remove a pet without selecting it!");
-    
-  }
 
-  // Fill out as part of Wave 3.
+  } else {
+    axios.delete(BASE_URL + selectedPetId)
+      .then(() => {
+        setResult(`Success! Pet #${selectedPetId} has been removed.`);
+      })
+      .catch((error) => {
+        setError(`${error.message}. Could not remove Pet #${selectedPetId}!`);
+    });
```

Comment:
to make it clear what the scope is on each line, this is how I would indent here:
```suggestion
    axios.delete(BASE_URL + selectedPetId)
    .then(() => {
      setResult(`Success! Pet #${selectedPetId} has been removed.`);
    })
    .catch((error) => {
      setError(`${error.message}. Could not remove Pet #${selectedPetId}!`);
    });
```